### ISSUE TITLE
Allow gatsby dev server to use auspice server 

### ISSUE THREAD

**Issue by eharkins** [nextstrain/nextstrain.org#158]
**Context + Description**  
It's not currently possible to run a gatsby dev server (to develop things like the splash page) and also have auspice serve and visualize datasets. That is, live development of the static site excludes any use of the rest of the site.

This means options for developing the static site are:
- run `cd static-site && gatsby develop` but not have the datasets, narratives be served by auspice
- run `npm run server` and rebuild gatsby each time you want to see changes to stuff in `static-site`

**Examples**  
This would make developing pages like https://github.com/nextstrain/nextstrain.org/pull/156 much easier where we are developing the static site's presentation of available sit rep narratives and so we need to be able to call `/charon/getAvailable`(which only comes from the nextstrain.org server, not gatsby) to get those narratives but also want to not have to rebuild the site to see our cosmetic changes.

**Possible solution**   
@jameshadfield said
> There are a couple of options worth exploring:
(1) Run both — gatsby dev server on port 8000 and npm run server on 5000 and try to get the API request from gatsby to go to port 5000. Is this possible? Not sure
(2) Mock the api call by replacing the fetch call in the React component with a hardcoded array which is similar to what the API would have returned. Should be pretty easy for a one-off.
(3) See if it’s possible to run the gatsby dev server from within the nextstrain.org server. I imagine this is hard.

Personally I'm using 2. at the moment but James notes that long term, implementing 1. would be 
> pretty cool as we do more and more development, and shouldn’t be too hard with a nginx like reverse proxy. But not really worth investing in time at the moment.

However, if there are gatsby pros out there reading this, have at it 🏄

**Comment by jameshadfield** [comment]
This is becoming a higher priority. 

A bit of history: There used to be a clear-cut separation between the "static-site" (built using gatsby) and the customised auspice client. The former could be developed using the gatsby dev server, while the latter could be developed [using the auspice dev server](https://github.com/nextstrain/nextstrain.org#testing-locally). 

As we've added functionality (e.g. authentication, a page which lists all the nCoV sit-reps), the "static-site" relies on API endpoints implemented in the nextstrain.org server, thus making the gatsby dev server inadequate. 

There's no hard necessity to have a hot-reloading dev-server, but it's quite a roadblock to future development.

**Comment by didithilmy** [comment]
Hello, I'm new here, but I think I might have a solution.

So, my approach is similar to option 1, which is running both the dev-server and gatsby dev server, but I take advantage of Gatsby's built-in development API proxy (https://www.gatsbyjs.com/docs/api-proxy/).

I made a proof-of-concept in my fork https://github.com/didithilmy/nextstrain.org/tree/use-gatsby-dev. Try to run `npm run dev` from the root directory.

Unfortunately, it involves Gatsby knowing what paths are NOT supposed to be served by Gatsby, which is located in `server.js`, outside `static-site`. In my solution, I moved the `auspicePaths` into a separate file, which is included by `static-site/gatsby-config.js`.

Hope this helps
